hangupsbot/plugins/cleverbot.py

```diff
@@ -251,66 +29,32 @@ def _handle_incoming_message(bot, event, command):
     percentage = bot.get_config_suboption(event.conv_id, 'cleverbot_percentage_replies')
 
     if randrange(0, 101, 1) < float(percentage):
-        text = yield from cleverbot_ask(event.conv_id, event.text)
-        if text:
-            yield from bot.coro_send_message(event.conv_id, text)
-
+        yield from chat(bot, event)
 
 def chat(bot, event, *args):
-    """chat to Cleverbot"""
+    """ Chat to Cleverbot. Usage: /bot chat hi cleverbot! """
+    if not cw:
+        api_key = bot.get_config_option("cleverbot_api_key")
+        if not api_key:
+            response = "API key not defined: config.cleverbot_api_key"
+            logger.error(response)
+            yield from bot.coro_send_message(event.conv_id, response)
+            return
+        else:
+            global cw
+            cw = CleverWrap(api_key)
     if args:
-        text = yield from cleverbot_ask(event.conv_id, ' '.join(args))
-        if not text:
-            text = _("<em>Cleverbot is silent</em>")
-
+        input_text = " ".join(args)
     else:
-        text = _("<em>you have to say something to Cleverbot</em>")
-
-    yield from bot.coro_send_message(event.conv_id, text)
+        input_text = event.text
+    response = cw.say(input_text)
```

Comment:
The plugin should preserve the [original behaviour](https://github.com/hangoutsbot/hangoutsbot/pull/750/files#diff-1811479853cf33f425ecb0217d11aba5L287) of the plugin it is based on. If there were a configurable option, then the default unconfigured behaviour would be a separate cleverbot chat session for each different conversation. This allows for the **principle of least surprise**:
* fulfills the expectation of users who have encountered the previous cb plugin
* privileged information will not be accidentally shared by the bot when used between different hangouts groups